src/node_crypto.cc

```diff
@@ -847,29 +847,42 @@ static unsigned long AddCertsFromFile(  // NOLINT(runtime/int)
   return err;
 }
 
+
+static void IsRootCertStoreEmpty(const FunctionCallbackInfo<Value>& args) {
+  return args.GetReturnValue().Set(root_cert_store == nullptr);
+}
+
+
+static void LoadExtraRootCertsFile(const FunctionCallbackInfo<Value>& args) {
+  ClearErrorOnReturn clear_error_on_return;
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!root_cert_store && !extra_root_certs_file.empty()) {
```

Comment:
Shouldn't this condition be split into two, like in the current code? If `root_cert_store` is already created and `extra_root_certs_file` is not empty, we should still add them, right?

Reply:
If root_cert_store is already created and extra_root_certs_file is not empty, we should still add them, right?
No, it seems the original codes won't set `root_cert_store` again if it's available. Please remind me if I mistake something.

Reply:
You are correct. I misread the code. But I still have one question. As per old code if the `root_cert_store` is not there yet, a new instance of `NewRootCertStore` is created and assigned to it. Shouldn't the new code also do the same?

Reply:
As per old code if the root_cert_store is not there yet, a new instance of NewRootCertStore is created and assigned to it.
Yes. One of the reasons is that I use `root_cert_store` to indicate whether the certs have been loaded in the flowing test. 

But I think you are right. I'm going to modify the code and make the new code do the same.